diff -ru a/configure.ac b/configure.ac
--- a/configure.ac	2015-01-05 22:41:44.536781042 +0400
+++ b/configure.ac	2015-01-05 23:04:18.026781687 +0400
@@ -701,17 +701,6 @@
     AC_MSG_ERROR([at least one API should be enabled])
 fi
 
-# Building OpenGL ES1 and/or ES2 without OpenGL is not supported on mesa 9.0.x
-if test "x$enable_opengl" = xno -a \
-        "x$enable_gles1" = xyes; then
-    AC_MSG_ERROR([Building OpenGL ES1 without OpenGL is not supported])
-fi
-
-if test "x$enable_opengl" = xno -a \
-        "x$enable_gles2" = xyes; then
-    AC_MSG_ERROR([Building OpenGL ES2 without OpenGL is not supported])
-fi
-
 AM_CONDITIONAL(HAVE_OPENGL, test "x$enable_opengl" = xyes)
 AM_CONDITIONAL(HAVE_OPENGL_ES1, test "x$enable_gles1" = xyes)
 AM_CONDITIONAL(HAVE_OPENGL_ES2, test "x$enable_gles2" = xyes)
@@ -1096,8 +1085,8 @@
 DRI_DIRS=''
 dnl Duplicates in DRI_DIRS are removed by sorting it at the end of this block
 if test -n "$with_dri_drivers"; then
-    if test "x$enable_opengl" != xyes; then
-        AC_MSG_ERROR([--with-dri-drivers requires OpenGL])
+    if test "x$enable_opengl" != xyes -a "x$enable_egl" != xyes; then
+        AC_MSG_ERROR([--with-dri-drivers requires OpenGL or EGL])
     fi
 
     dri_drivers=`IFS=', '; echo $with_dri_drivers`
@@ -1451,11 +1440,9 @@
 AC_SUBST([LLVM_VERSION])
 AC_SUBST([CLANG_RESOURCE_DIR])
 
-case "x$enable_opengl$enable_gles1$enable_gles2" in
-x*yes*)
+if test "x$enable_opengl" = xyes; then
     EGL_CLIENT_APIS="$EGL_CLIENT_APIS "'$(GL_LIB)'
-    ;;
-esac
+fi
 
 AC_SUBST([VG_LIB_DEPS])
 AC_SUBST([EGL_CLIENT_APIS])
diff -ru a/src/gallium/targets/egl-static/Makefile.am b/src/gallium/targets/egl-static/Makefile.am
--- a/src/gallium/targets/egl-static/Makefile.am	2014-09-19 21:03:22.000000000 +0400
+++ b/src/gallium/targets/egl-static/Makefile.am	2015-01-05 23:01:04.576781596 +0400
@@ -113,7 +113,7 @@
 if HAVE_OPENGL
 AM_CPPFLAGS += \
 	-I$(top_srcdir)/src/mesa \
-	-DFEATURE_GL=1
+	$(API_DEFINES)
 
 egl_gallium_la_LIBADD += \
 	$(top_builddir)/src/mesa/libmesagallium.la
@@ -140,16 +140,6 @@
 endif
 endif
 
-if HAVE_OPENGL_ES1
-AM_CPPFLAGS += \
-	-DFEATURE_ES1=1
-endif
-
-if HAVE_OPENGL_ES2
-AM_CPPFLAGS += \
-	-DFEATURE_ES2=1
-endif
-
 if HAVE_OPENVG
 AM_CPPFLAGS += \
 	-I$(top_srcdir)/src/gallium/state_trackers/vega \
diff -ru a/src/mesa/Makefile.am b/src/mesa/Makefile.am
--- a/src/mesa/Makefile.am	2014-09-19 21:03:23.000000000 +0400
+++ b/src/mesa/Makefile.am	2015-01-05 23:05:53.516781733 +0400
@@ -33,8 +33,10 @@
 SUBDIRS += drivers/osmesa
 endif
 
+if HAVE_OPENGL
 gldir = $(includedir)/GL
 gl_HEADERS = $(top_srcdir)/include/GL/*.h
+endif
 
 .PHONY: $(BUILDDIR)main/git_sha1.h.tmp
 $(BUILDDIR)main/git_sha1.h.tmp:
@@ -146,8 +148,10 @@
 	main/streaming-load-memcpy.c
 libmesa_sse41_la_CFLAGS = $(AM_CFLAGS) -msse4.1
 
+if HAVE_OPENGL
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gl.pc
+endif
 
 $(BUILDDIR)program/lex.yy.c: program/program_lexer.l
 	$(AM_V_at)$(MKDIR_P) program
