--- common/Makefile
+++ common/Makefile
@@ -233,7 +233,7 @@
 CPPFLAGS += -I..
 
 $(obj)env_embedded.o: $(src)env_embedded.c
-	$(CC) $(AFLAGS) -Wa,--no-warn \
+	$(CROSS_COMPILE)$(CC) $(AFLAGS) -Wa,--no-warn \
 		-DENV_CRC=$(shell $(obj)../tools/envcrc) \
 		-c -o $@ $(src)env_embedded.c
 
--- config.mk
+++ config.mk
@@ -86,7 +86,7 @@
 HOSTCFLAGS	+= $(call os_x_before, 10, 4, "-traditional-cpp")
 HOSTLDFLAGS	+= $(call os_x_before, 10, 5, "-multiply_defined suppress")
 else
-HOSTCC		= gcc
+HOSTCC		= $(CHOST)-gcc
 endif
 
 ifeq ($(HOSTOS),cygwin)
